sparql/parse.go

```diff
@@ -25,11 +25,18 @@ func (query *Query) ExtractEntities(src string) ([]string, error) {
 
 				pos = strings.Index(string(tripleBlock[startPos:]), " .") + startPos // We need to add our currert start position as index starts from 0
 				for pos > startPos {
-					aTriple := query.NewTriple(tripleBlock[startPos:pos], "")
-					entities = appendEntity(entities, aTriple.Subject)
-					startPos = pos + 2
-					pos = strings.Index(tripleBlock[startPos:], " .") + startPos // We need to add our currert start position as index starts from 0
+					// This check must happen in order to determine if there is literal block
+					// that may match our end-of-statement marker.
+					if strings.Count(string(tripleBlock[startPos:pos]), "\"")%2 != 0 {
```

Comment:
What happens if there's a single `"` inside a literal?  Will this still work? I'm wondering if we need to use goyacc or something to generate a real parser otherwise we're going to keep piling hacks on hacks.

Reply:
I have the same concern. Wasn't aware of goyacc, I'll take a look! I do think this particular is a bit of an outlier (for now) so there doesn't seem to be a big rush in merging this just yet. 

I will investigate.